File: backend/core/routes/module.py
from fastapi import APIRouter, Depends, HTTPException, Request

from core.utils import ESException, get_escapy


# Router /module
router = APIRouter(tags=["Module"])



# Get all available modules list
@router.get("/list")
def get_modules_list(escapy=Depends(get_escapy)):
  try:
    return escapy.get_modules_list()
  except Exception as e:
    raise ESException(e)


# Get module options
@router.get("/options")
def get_module_options(name: str, escapy=Depends(get_escapy)):
  try:
    return escapy.get_module_options(name)
  except Exception as e:
    raise ESException(e)


# Module public files
@router.get("/public/{name}/{path:path}")
def get_module_public_file(name: str, path: str, escapy=Depends(get_escapy)):
  try:
    module = escapy.tasks.modules.get_module(name)

    return module.get_public_file(path)
  except Exception as e:
    raise ESException(e)

File: frontend/src/api/module.js
import { api, handleError } from "./config";

// Get modules list
export async function getModulesList() {
  try {
    const { data } = await api.get("/module/list");

    return data;
  } catch (err) {
    handleError(err);
  }
}

// Get module options
export async function getModuleOptions(name) {
  try {
    const { data } = await api.get("/module/options", {
      params: { name }
    });

    return data;
  } catch (err) {
    handleError(err);
  }
}
